src/api/reports/stats.py

```diff
@@ -0,0 +1,76 @@
+from api.models import FragmentStats, RestorationStats, SCLStats, SurveyStats
+
+
+def _calc_summary(data):
+    num_landscapes = 0
+    habitat_area = 0
+    protected_area = 0
+
+    for rec in data:
+        num_landscapes += 1
+        habitat_area += rec.area
+        biome_areas = rec.biome_areas or []
+        protected_area += sum([ba.get("protected") or 0 for ba in biome_areas])
+
+    try:
+        percent_protected_area = float(protected_area) / float(habitat_area)
+    except (TypeError, ValueError, ZeroDivisionError):
+        percent_protected_area = None
+
+    return dict(
+        num_landscapes=num_landscapes,
+        habitat_area=habitat_area,
+        protected_area=protected_area,
+        percent_protected_area=percent_protected_area,
+    )
+
+
+def calc_landscape_stats(country, date, species):
+    return dict(
+        conservation_landscape_stats=calc_conservation_landscape_stats(
+            country, date, species
+        ),
+        fragment_landscape_stats=calc_fragment_landscape_stats(country, date, species),
+        restoration_landscape_stats=calc_restoration_landscape_stats(
+            country, date, species
+        ),
+        survey_landscape_stats=calc_survey_landscape_stats(country, date, species),
+    )
+
+
+def calc_conservation_landscape_stats(country, date, species):
```

Comment:
Could conceivably dry some of this up, but not important

Reply:
I thought about DRYing up more but thought it was easier to read